source/fe/fe_poly_tensor.cc

```diff
@@ -84,66 +83,82 @@ namespace internal
                 const unsigned int nn = cell->neighbor_face_no(f);
 
                 if (nn < GeometryInfo<dim>::faces_per_cell / 2)
-                  for (unsigned int j = 0; j < dofs_per_face; ++j)
+                  for (unsigned int j = 0; j < fe.dofs_per_face; ++j)
                     {
-                      Assert(f * dofs_per_face + j < face_sign.size(),
+                      const unsigned int cell_j = fe.face_to_cell_index(j, f);
+                      // something in FiniteElement or FiniteElementData base
+                      // classes
+
+                      Assert(f * fe.dofs_per_face + j < face_sign.size(),
+                             ExcInternalError());
+                      Assert(mapping_type.size() == 1 ||
+                               cell_j < mapping_type.size(),
```

Comment:
I think this check and the one below would be clearer if we just added
```cpp
const bool one_mapping = mapping_type.size() == 1;
```
(or some other similar name) at the top of the function.

Reply:
Good plan. These convenience functions- I suppose they should be `const` functions and `protected` privacy, right? Let me know if I did anything silly with this part.

Unfortunately, however, since these sign change functions are not part of the FE_PolyTensor class itself, the convenience functions will not work in this context.

Reply:
Good point. I should read the whole diff.